src/code/z_parameter.c

```diff
@@ -4634,8 +4634,20 @@ void Interface_DrawClock(PlayState* play) {
                               R_THREE_DAY_CLOCK_SUN_MOON_CUTOFF * 4.0f);
 
             // determines the current hour
-            for (sp1C6 = 0; sp1C6 <= 24; sp1C6++) {
+            for (sp1C6 = 0; sp1C6 < ARRAY_COUNT(sThreeDayClockHours) - 1; sp1C6++) {
+                //! @bug In the original game, this loop iterates over an array of clock hour
+                // values to determine what the current hour is which is used to index into a
+                // texture pointer array. When the loop reaches the last value, the clock is
+                // actually equal to this value for a frame or two before it rolls over.
+                // Because this check is < and not <=, it will actually iterate past it by one
+                // due to the for loop terminating. This results in 25, which is OOB for the
+                // sThreeDayClockHourTextures[] read later. On console, this results in the hour
+                // disappearing for a frame or two between 11 changing to 12.
```

Comment:
```suggestion
                //! @bug When this loop iterates to the end without a break, this results in `sp1C6` being 25, leading to an OOB read for sThreeDayClockHourTextures[].
                // In practice, this occurs for two frames between changing hour 23 to 24.
```
I think this comment can be summized much more briefly. What do you think of this?

Reply:
I would still like to mention that the "fix" should be <= instead of < somewhere even if we dont have a define for it. I think it's clear what the dev meant.